__tests__/distributors/local-installer.test.ts

```diff
@@ -0,0 +1,194 @@
+      javaVersion: actualJavaVersion,
+      javaPath: path.join('Java_LocalJDKFile_jdk', inputs.version, inputs.arch)
+    };
+
+    mockJavaBase = new LocalDistributor(inputs, jdkFile);
+    expected.javaPath = path.join('Java_LocalJDKFile_jdk', inputs.version, inputs.arch);
+    await expect(mockJavaBase.setupJava()).resolves.toEqual(expected);
+    expect(spyTcFind).toHaveBeenCalled();
+    expect(spyCoreInfo).toHaveBeenCalledWith(`Resolved Java ${actualJavaVersion} from tool-cache`);
+    expect(spyCoreInfo).not.toHaveBeenCalledWith(
+      `Java ${inputs.version} is not found in tool-cache. Trying to unpack JDK file...`
+    );
+  });
+
+  it('java is unpacked from jdkfile', async () => {
+    const inputs = { version: '11.0.289', arch: 'x86', packageType: 'jdk' };
+    const jdkFile = expectedJdkFile;
+    const expected = {
+      javaVersion: '11.0.289',
+      javaPath: path.join('Java_LocalJDKFile_jdk', inputs.version, inputs.arch)
+    };
+
+    mockJavaBase = new LocalDistributor(inputs, jdkFile);
+    expected.javaPath = path.join('Java_LocalJDKFile_jdk', inputs.version, inputs.arch);
+    await expect(mockJavaBase.setupJava()).resolves.toEqual(expected);
+    expect(spyTcFind).toHaveBeenCalled();
+    expect(spyCoreInfo).not.toHaveBeenCalledWith(
+      `Resolved Java ${actualJavaVersion} from tool-cache`
+    );
+    expect(spyCoreInfo).toHaveBeenCalledWith(`Extracting Java from '${jdkFile}'`);
+    expect(spyCoreInfo).toHaveBeenCalledWith(
+      `Java ${inputs.version} is not found in tool-cache. Trying to unpack JDK file...`
+    );
+  });
+
+  it('jdk file is not found', async () => {
+    const inputs = { version: '11.0.289', arch: 'x86', packageType: 'jdk' };
+    const jdkFile = 'not_existing_one';
+    const expected = {
+      javaVersion: '11.0.289',
+      javaPath: path.join('Java_LocalJDKFile_jdk', inputs.version, inputs.arch)
+    };
+
+    mockJavaBase = new LocalDistributor(inputs, jdkFile);
+    expected.javaPath = path.join('Java_LocalJDKFile_jdk', inputs.version, inputs.arch);
+    await expect(mockJavaBase.setupJava()).rejects.toThrowError(
+      "JDK file is not found in path 'not_existing_one'"
+    );
+    expect(spyTcFind).toHaveBeenCalled();
+    expect(spyCoreInfo).not.toHaveBeenCalledWith(
+      `Resolved Java ${actualJavaVersion} from tool-cache`
+    );
+    expect(spyCoreInfo).not.toHaveBeenCalledWith(`Extracting Java from '${jdkFile}'`);
+    expect(spyCoreInfo).toHaveBeenCalledWith(
+      `Java ${inputs.version} is not found in tool-cache. Trying to unpack JDK file...`
+    );
+  });
+
+  it.each([
+    [{ version: '8.0.289', arch: 'x64', packageType: 'jdk' }, 'otherJdkFile'],
+    [{ version: '11.0.289', arch: 'x64', packageType: 'jdk' }, 'otherJdkFile'],
+    [{ version: '12.0.289', arch: 'x64', packageType: 'jdk' }, 'otherJdkFile'],
+    [{ version: '11.1.11', arch: 'x64', packageType: 'jdk' }, 'not_existing_one']
+  ])('inputs %s, jdkfile %s', async (inputs, jdkFile) => {
```

Comment:
test name is unclear, what is expected, what do you check in this test?

Reply:
Done. Thank you.